index.d.ts

```diff
@@ -0,0 +1,46 @@
+export type CurriedHandler<P = any, S = any> = (payload: P) => (state: S) => S;
+
+// TODO: figure out how import this from redux, since this pkg has no dependencies
+type Reducer = (state: any, action: {type: any, payload: any}) => any;
```

Comment:
`action.type` can be forced to `string`, I think. At least in the context warped-reducers.

Reply:
Actually if this is the case we can make the returned reducer's type include the action types it handles and the payload it handles:
```typescript
typeof createReducer ('App') ({
  loadData: noopAction,
  setData: (name: string) => (state: any) => ({...state, app: {data: name}}),
});
// > (state: any, action:
// {type: 'Warped/App/loadData', payload: any} |
// {type: 'Warped/App/setData', payload: string}) => any
```

But in practice it can be called with an action it'll ignore, and it's a valid call, c.f. noopAction.

So in order to actually reflect the runtime behavior it needs to be `any` or have `action: ... | any` in that type, which makes it useless...
Question is whether we make type declarations as lenient as untyped usage or not.